src/bin/build_octree.rs

```diff
@@ -122,50 +128,56 @@ fn should_split_node(node: &SplittedNode, resolution: f64) -> bool {
 
 fn split_node<'a, 'b: 'a, P>(
     scope: &Scope<'a>,
-    output_directory: &'b Path,
-    resolution: f64,
-    splitted_node: &SplittedNode,
+    build_meta: &'a BuildMeta,
+    node_id: &octree::NodeId,
     stream: P,
-    leaf_nodes_sender: &mpsc::Sender<octree::Node>,
+    leaf_nodes_sender: &mpsc::Sender<octree::NodeId>,
 ) where
     P: InternalIterator,
 {
-    let children = split(output_directory, resolution, &splitted_node.node, stream);
-    let (leaf_nodes, split_nodes): (Vec<_>, Vec<_>) = children
-        .into_iter()
-        .partition(|n| !should_split_node(n, resolution));
-
-    for child in split_nodes {
+    let (leaf_nodes, split_nodes) = split(build_meta, &node_id, stream);
+    for child_id in split_nodes {
         let leaf_nodes_sender_clone = leaf_nodes_sender.clone();
         scope.recurse(move |scope| {
-            let stream = octree::NodeIterator::from_disk(output_directory, &child.node.id).unwrap();
+            let bounding_cube = child_id.bounding_cube(&build_meta.root_bounding_cube);
```

Comment:
Rename the function into find_bounding_cube? It is not a getter, but does some computation.